cli/command/container/stop.go

```diff
@@ -39,16 +40,45 @@ func NewStopCommand(dockerCli *command.DockerCli) *cobra.Command {
 
 func runStop(dockerCli *command.DockerCli, opts *stopOptions) error {
 	ctx := context.Background()
+	timeout := time.Duration(opts.time) * time.Second
 
+	var waitAll sync.WaitGroup
 	var errs []string
+	const maxParallel int32 = 50
+
+	// make sure result is printed in order
+	output := map[string]chan error{}
 	for _, container := range opts.containers {
-		timeout := time.Duration(opts.time) * time.Second
-		if err := dockerCli.Client().ContainerStop(ctx, container, &timeout); err != nil {
-			errs = append(errs, err.Error())
-		} else {
-			fmt.Fprintf(dockerCli.Out(), "%s\n", container)
+		output[container] = make(chan error, 1)
+	}
+	waitAll.Add(1)
+	go func() {
+		for _, container := range opts.containers {
+			if err := <-output[container]; err != nil {
+				errs = append(errs, err.Error())
+			} else {
+				fmt.Fprintf(dockerCli.Out(), "%s\n", container)
+			}
 		}
+		waitAll.Done()
+	}()
```

Comment:
Ah, that's a good point. but I also feel like if we want to provide incremental progress it would be better to just use a single channel and print as they finish.

I know @tonistiigi asked for consistent ordered output, but I'm not sure I agree that's correct in this case. How strongly do you feel about the consistent ordering @tonistiigi  ?